entrypoint.sh

```diff
@@ -142,11 +142,12 @@ nginxConfiguration() {
     echo "Executing nginx configuration ..."
     if [ "$DISABLE_HTTPS" == "True" ] || [ "$DISABLE_HTTPS" == "true" ]; then
         echo "Disabling https in nginx."
-        mv -f /opt/files/nginx/zulip-enterprise-http /etc/nginx/sites-enabled/zulip-enterprise
+        crudini --set /etc/zulip/zulip.conf application_server http_only true
     fi
     sed -i "s/worker_processes .*/worker_processes $NGINX_WORKERS;/g" /etc/nginx/nginx.conf
     sed -i "s/client_max_body_size .*/client_max_body_size $NGINX_MAX_UPLOAD_SIZE;/g" /etc/nginx/nginx.conf
     sed -i "s/proxy_buffering .*/proxy_buffering $NGINX_PROXY_BUFFERING;/g" /etc/nginx/zulip-include/proxy_longpolling
+    /home/zulip/deployments/current/scripts/zulip-puppet-apply -f
```

Comment:
Is this needed? The image should start immediately and not have to run puppet apply again

Reply:
The changes to the nginx conf from the `http_only` flag don't get applied without running this.

Reply:
Still I don't want puppet to be run again in the "live" image. Can you check for other solutions, e.g. `sed` or so?

Reply:
Basically, the `DISABLE_HTTPS` flag now sets a `http_only` config in the `/etc/zulip/zulip.conf` which rewrites the nginx conf to listen on port 80 instead of 443. Previously this was done using the `setup_files` stuff.

Reply:
It only takes a couple of seconds to run, even less when there are no changes from the existing config, and it depends on the environment variables, so we have to put it in entrypoint.sh.

I can make it so that it only runs the puppet if there is some changes expected, but that will be a lot of code for what is a very minor increase in load time.

Reply:
@galexrt yeah, the problem is that the results of running `puppet` in this case depend on the user's configuration in `/etc/zulip/zulip.conf`.  What we'll want to do is check if the hash of `/etc/zulip/zulip.conf` has changed since the last time the image booted, and only run `puppet` if it has.  Since it's only a couple seconds of runtime, this is more a performance optimization than a correctness issue.  

I opened #137 for that follow-up work.

Reply:
I did a small tweak to move the `zulip-puppet-apply` call inside the if statement, so this now only affects users doing the `DISABLE_HTTPS` option.